lib/babe/parachain_inherents.go

```diff
@@ -0,0 +1,240 @@
+package babe
+
+import (
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// ValidityAttestation is an implicit or explicit attestation to the validity of a parachain
+// candidate.
+type ValidityAttestation scale.VaryingDataType
+
+// Implicit is for implicit attestation.
+type Implicit ValidatorSignature
+
+// Index Returns VDT index
+func (im Implicit) Index() uint {
+	return 1
+}
+
+// Explicit is for explicit attestation.
+type Explicit ValidatorSignature
+
+// Index Returns VDT index
+func (ex Explicit) Index() uint {
+	return 2
+}
+
+// GetValidityAttestation returns a ValidityAttestation varying data type.
+func GetValidityAttestation() ValidityAttestation {
+	vdt, err := scale.NewVaryingDataType(Implicit{}, Explicit{})
+	if err != nil {
+		panic(err)
+	}
+
+	return ValidityAttestation(vdt)
+}
+
+// DisputeStatement is a statement about a candidate, to be used within the dispute
+// resolution process. Statements are either in favour of the candidate's validity
+// or against it.
+type DisputeStatement scale.VaryingDataType
+
+// Valid is for a valid statement
+type Valid ValidDisputeStatementKind
+
+// Index Returns VDT index
+func (v Valid) Index() uint {
+	return 0
```

Comment:
Usually this should start at `1` I believe.  But I did notice that in the polkadot code it had this macro above the definition:
```
#[codec(index = 0)]
Valid(ValidDisputeStatementKind),
```
Can you verify that that it does use the index when serialising on the rust side? We should probably have a test for the whole nested `VaryingDataType` just to ensure this is working as expected.

Reply:
done